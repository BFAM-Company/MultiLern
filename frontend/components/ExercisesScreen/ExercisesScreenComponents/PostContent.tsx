import React, { useEffect, useRef, useState } from 'react'
import { Animated, Dimensions, Image, ImageBackground, Modal, ScrollView, StyleSheet, Text, TouchableOpacity, View } from 'react-native'
import Carousel from 'react-native-snap-carousel';
import Comments from './Comments';



interface PostContentProps{
    id:number,
    title: string,
    user_data: any[],
    date: string,
    content: string,
    handleClose: any,
    posts_images: any[]
}


function PostContent({id, title, handleClose, posts_images, content, user_data, date}: PostContentProps) {
    const windowWidth = Dimensions.get('window').width;
    const windowHeight = Dimensions.get('window').height;
    const [imageZoom, setImageZoom] = useState<boolean>(false)
    const [zoomedImage, setZoomedImage] = useState<string>('');

    useEffect(()=>{
        //console.log(JSON.stringify(posts_images))
    },[])

    const _renderItem = ({item, index}: any) => {
        return (
            <TouchableOpacity
                style={styles.imageContainer}
                activeOpacity={1}
                onPress={()=>{
                    setZoomedImage(item.images.img)
                    setImageZoom(true)
                }}
                testID={`image-touchable-${index}`}
            >
                <Image 
                    key={index}
                    source={{uri: item.images.img}}
                    style={styles.postImage}
                />
            </TouchableOpacity>
        );
    }
  return (
    <View        
        style={styles.fixedContainerBgc}
        >   
            <ScrollView
                onScroll={handleClose}
            >
                <View style={styles.separator}></View>
                <View style={styles.contentContainer}>
                    <View
                        style={{width:windowWidth, display:'flex', alignItems:'center', justifyContent:'center'}}
                    >
                        <View style={{width:100, height:3, backgroundColor:'lightgray', marginTop:5, marginBottom:50}}></View>
                    </View>
                    <View style={styles.userInfoContainer}>
                        <Image
                            style={styles.userIcon}
                            source={user_data[0].users.avatar?{uri: user_data[0].users.avatar}:require('./../../../assets/demo-user-icon.png')}
                        />
                        <View style={styles.verticalContainer}>
                            <Text
                                style={styles.nickname}
                            >{user_data[0].users.nickname}</Text>
                            <Text
                                style={styles.date}
                            >{date}</Text>
                        </View>
                    </View>
                    <Carousel
                        ref={(c) => { let _carousel = c }}
                        data={posts_images}
                        renderItem={_renderItem}
                        sliderWidth={windowWidth}
                        itemWidth={windowWidth*0.8}
                        layout={'stack'} 
                        layoutCardOffset={20} 
                    />
                    <Text 
                        numberOfLines={4}
                        style={styles.postTitle}
                    >
                        {title}
                    </Text>
                    <Text
                        style={styles.contentText}
                    >
                        {content}
                    </Text>
                    <Comments id={id} />
                    {/* <View style={{height:100}}></View> */}
                </View>
            </ScrollView>
            <Modal
                visible={imageZoom}
                testID='zoom-modal'
            >
                <TouchableOpacity
                    onPress={()=>{setImageZoom(false)}}
                >
                    <Image
                        testID='zoomed-image'
                        source={{uri: zoomedImage}}
                        style={{width:windowWidth, height:windowHeight, resizeMode:'contain'}}
                    />
                    {/* <ImageViewer imageUrls={[{url: zoomedImage}]} /> */}
                </TouchableOpacity>
            </Modal>
        </View>
  );
}




const styles = StyleSheet.create({
    container:{
        width:'100%',
        height:'100%',
        display:'flex',
        flexDirection:'row',
        alignItems:'center',
        justifyContent:'space-around'
    },
    fixedContainerBgc:{
        width:'100%',
        backgroundColor:'transparent'
    },
    separator:{
        width:'100%',
        height:250,
    },
    contentContainer:{
        width:'100%',
        marginTop:-50,
        height:'auto',
        //paddingBottom:100,
        display: 'flex',
        alignItems: 'center',
        justifyContent:'flex-start',
        borderTopRightRadius:50,
        borderTopLeftRadius:50,
        backgroundColor:'#fff',
        shadowColor: "#000",
        shadowOffset: {
                width: 0,
                height: 0,
        },
        shadowOpacity: 0.44,
        shadowRadius: 10.32,

        elevation: 16,
    },
    postImage:{
        width:'100%',
        height:300,
        resizeMode:'cover',
        borderRadius:10,
    },
    userIcon:{
        width:50,
        height:50,
        borderRadius:100,
      },
    postTitle:{
        fontSize:30,
        fontWeight:'900',
        paddingTop:50,
        margin:20,
        paddingRight:10,
        color:'black'
    },
    imageContainer:{
        width:'auto',
        height:'auto'
    },
    contentText:{
        fontSize:22,
        marginTop:50,
        margin:20,
        fontWeight:'700',
        color:'rgb(45,45,55)'
    },
    userInfoContainer:{
        width:'100%',
        display:'flex',
        flexDirection:'row',
        justifyContent:'flex-start',
        alignItems:'center',
        padding:20,
        marginBottom:10,
    },
    verticalContainer:{
        display:'flex',
        flexDirection:'column',
        alignItems:'flex-start',
        justifyContent:'space-around',
        marginLeft:40,
    },
    nickname:{
        fontSize:22,
        fontWeight:'700',
        color:'rgb(45,45,55)'
    },
    date:{
        fontSize:16,
        fontWeight:'500',
        color:'gray',
    }
})

export default PostContent;