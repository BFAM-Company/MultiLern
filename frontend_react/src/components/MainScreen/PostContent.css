.post-content-container {
  width: 100%;
  background-color: transparent;
}

.scroll-container {
  overflow-y: auto;
}

.separator {
  width: 100%;
  height: 250px;
}

.content-container {
  width: 100%;
  margin-top: -50px;
  padding-top: 20px;
  padding-bottom: 50px;
  border-top-right-radius: 50px;
  border-top-left-radius: 50px;
  background-color: #fff;
  box-shadow: 0px 4px 10px rgba(0, 0, 0, 0.25);
  display: flex;
  flex-direction: column;
  align-items: center;
}

.drag-indicator {
  width: 100px;
  height: 3px;
  background-color: lightgray;
  margin: 10px 0 50px;
}

.user-info-container {
  width: 100%;
  display: flex;
  align-items: center;
  padding: 20px;
  margin-bottom: 10px;
}

.user-icon {
  width: 50px;
  height: 50px;
  border-radius: 50%;
  object-fit: cover;
}

.user-info {
  margin-left: 20px;
}

.nickname {
  font-size: 1.2rem;
  font-weight: bold;
  color: #2d2d37;
}

.date {
  font-size: 0.9rem;
  color: gray;
}

.carousel-container {
  width: 80%;
  margin-bottom: 20px;
}

.image-container {
  border: none;
  background: none;
  padding: 0;
  cursor: pointer;
}

.post-image {
  width: 100%;
  height: 300px;
  object-fit: cover;
  border-radius: 10px;
}

.post-title {
  font-size: 1.8rem;
  font-weight: bold;
  color: black;
  margin: 20px;
}

.content-text {
  font-size: 1.1rem;
  font-weight: bold;
  color: rgb(45, 45, 55);
  margin: 20px;
}

.modal {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.8);
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
}

.zoomed-image {
  max-width: 90%;
  max-height: 90%;
  object-fit: contain;
}
