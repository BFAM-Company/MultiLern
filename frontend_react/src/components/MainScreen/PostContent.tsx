import React, { useEffect, useState } from 'react';
import './PostContent.css';
import Carousel from 'react-multi-carousel';
import 'react-multi-carousel/lib/styles.css';

interface PostContentProps {
  id: number;
  title: string;
  user_data: any[];
  date: string;
  content: string;
  handleClose: () => void;
  posts_images: any[];
}

const PostContent: React.FC<PostContentProps> = ({ id, title, handleClose, posts_images, content, user_data, date }) => {
  const [imageZoom, setImageZoom] = useState<boolean>(false);
  const [zoomedImage, setZoomedImage] = useState<string>('');

  useEffect(() => {
    // Perform any necessary setup or cleanup
  }, []);

  const handleImageClick = (img: string) => {
    setZoomedImage(img);
    setImageZoom(true);
  };

  return (
    <div className="post-content-container">
      <div className="scroll-container" onScroll={handleClose}>
        <div className="separator"></div>
        <div className="content-container">
          <div className="drag-indicator"></div>
          <div className="user-info-container">
            <img
              className="user-icon"
              src={user_data[0]?.users?.avatar || '/path-to-demo-user-icon.png'}
              alt="User Avatar"
            />
            <div className="user-info">
              <span className="nickname">{user_data[0]?.users?.nickname}</span>
              <span className="date">{date}</span>
            </div>
          </div>

          <Carousel
            additionalTransfrom={0}
            arrows
            autoPlay={false}
            centerMode={false}
            containerClass="carousel-container"
            draggable
            focusOnSelect={false}
            infinite
            itemClass="carousel-item"
            keyBoardControl
            minimumTouchDrag={80}
            renderDotsOutside={false}
            responsive={{
              desktop: {
                breakpoint: { max: 3000, min: 1024 },
                items: 1,
              },
              tablet: {
                breakpoint: { max: 1024, min: 464 },
                items: 1,
              },
              mobile: {
                breakpoint: { max: 464, min: 0 },
                items: 1,
              },
            }}
            showDots
            sliderClass=""
            slidesToSlide={1}
            swipeable
          >
            {posts_images.map((item, index) => (
              <button
                key={index}
                className="image-container"
                onClick={() => handleImageClick(item.images.img)}
                data-testid={`image-touchable-${index}`}
              >
                <img
                  src={item.images.img}
                  alt={`Post ${index + 1}`}
                  className="post-image"
                />
              </button>
            ))}
          </Carousel>

          <h2 className="post-title">{title}</h2>
          <p className="content-text">{content}</p>
          <Comments id={id} />
        </div>
      </div>

      {imageZoom && (
        <div className="modal" onClick={() => setImageZoom(false)} data-testid="zoom-modal">
          <img
            src={zoomedImage}
            alt="Zoomed"
            className="zoomed-image"
            data-testid="zoomed-image"
          />
        </div>
      )}
    </div>
  );
};

export default PostContent;
