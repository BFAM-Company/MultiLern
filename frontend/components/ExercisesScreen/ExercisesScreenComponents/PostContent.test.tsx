import React from 'react'
import { fireEvent, render } from "@testing-library/react-native";
import PostContent from "./PostContent";

jest.mock("@react-native-async-storage/async-storage", () => require("../../../__mocks__/mock-async-storage"));

describe('PostContent component', () => {
  const mockUserData = [
    {
      users: {
        avatar: 'mock-avatar-url',
        nickname: 'JohnDoe'
      }
    }
  ];
  const mockProps = {
    id: 1,
    title: 'Test Post',
    user_data: mockUserData,
    date: '2024-05-29',
    content: 'This is a test post content',
    handleClose: jest.fn(),
    posts_images: [{ images: { img: 'mock-image-url' } }]
  };

  it('renders correctly', () => {
        const { toJSON } = render(<PostContent {...mockProps} />);
        expect(toJSON()).toMatchSnapshot();
    });

    it('opens zoom modal with correct image when an image is pressed', () => {
    const { getByTestId } = render(<PostContent {...mockProps} />);
    const imageTouchable = getByTestId('image-touchable-0');
    fireEvent.press(imageTouchable);

    const zoomedImage = getByTestId('zoomed-image');
    expect(zoomedImage.props.source.uri).toBe('mock-image-url');
  });

  it('closes the zoom modal when the modal is pressed', () => {
    const { getByTestId } = render(<PostContent {...mockProps} />);
    const imageTouchable = getByTestId('image-touchable-0');
    fireEvent.press(imageTouchable);

    const zoomModal = getByTestId('zoom-modal');
    fireEvent.press(zoomModal);

    expect(zoomModal.props.visible).toBe(true);
  });

});