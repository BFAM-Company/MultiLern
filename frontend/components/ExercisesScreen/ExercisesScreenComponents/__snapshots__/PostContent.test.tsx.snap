// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostContent component renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "transparent",
      "width": "100%",
    }
  }
  testID="post-content"
>
  <RCTScrollView
    onScroll={[MockFunction]}
  >
    <View>
      <View
        style={
          {
            "height": 250,
            "width": "100%",
          }
        }
      />
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderTopLeftRadius": 50,
            "borderTopRightRadius": 50,
            "display": "flex",
            "elevation": 16,
            "height": "auto",
            "justifyContent": "flex-start",
            "marginTop": -50,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.44,
            "shadowRadius": 10.32,
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "width": 750,
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "lightgray",
                "height": 3,
                "marginBottom": 50,
                "marginTop": 5,
                "width": 100,
              }
            }
          />
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "marginBottom": 10,
              "padding": 20,
              "width": "100%",
            }
          }
        >
          <Image
            source={
              {
                "uri": "mock-avatar-url",
              }
            }
            style={
              {
                "borderRadius": 100,
                "height": 50,
                "width": 50,
              }
            }
          />
          <View
            style={
              {
                "alignItems": "flex-start",
                "display": "flex",
                "flexDirection": "column",
                "justifyContent": "space-around",
                "marginLeft": 40,
              }
            }
          >
            <Text
              style={
                {
                  "color": "rgb(45,45,55)",
                  "fontSize": 22,
                  "fontWeight": "700",
                }
              }
            >
              JohnDoe
            </Text>
            <Text
              style={
                {
                  "color": "gray",
                  "fontSize": 16,
                  "fontWeight": "500",
                }
              }
            >
              2024-05-29
            </Text>
          </View>
        </View>
        <RCTScrollView
          activeAnimationOptions={null}
          activeAnimationType="timing"
          activeSlideAlignment="center"
          activeSlideOffset={20}
          apparitionDelay={0}
          automaticallyAdjustContentInsets={false}
          autoplay={false}
          autoplayDelay={1000}
          autoplayInterval={3000}
          callbackOffsetMargin={5}
          collapsable={false}
          containerCustomStyle={{}}
          contentContainerCustomStyle={{}}
          contentContainerStyle={
            [
              {
                "paddingLeft": 75,
                "paddingRight": 75,
              },
              {},
            ]
          }
          data={
            [
              {
                "images": {
                  "img": "mock-image-url",
                },
              },
            ]
          }
          decelerationRate="fast"
          directionalLockEnabled={true}
          enableMomentum={false}
          enableSnap={true}
          firstItem={0}
          hasParallaxImages={false}
          horizontal={true}
          inactiveSlideOpacity={0.7}
          inactiveSlideScale={0.9}
          inactiveSlideShift={0}
          inverted={false}
          itemWidth={600}
          layout="stack"
          layoutCardOffset={20}
          lockScrollTimeoutDuration={1000}
          lockScrollWhileSnapping={false}
          loop={false}
          loopClonesPerSide={3}
          onLayout={[Function]}
          onResponderRelease={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          overScrollMode="never"
          pinchGestureEnabled={false}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEnabled={true}
          scrollEventThrottle={1}
          scrollsToTop={false}
          shouldOptimizeUpdates={true}
          showsHorizontalScrollIndicator={false}
          showsVerticalScrollIndicator={false}
          slideStyle={{}}
          sliderWidth={750}
          style={
            {
              "flexDirection": "row",
              "opacity": 0,
              "width": 750,
            }
          }
          swipeThreshold={20}
          useScrollView={false}
          vertical={false}
        >
          <View>
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                {
                  "opacity": 1,
                  "transform": [
                    {
                      "scale": 1,
                    },
                    {
                      "translateX": 0,
                    },
                  ],
                  "width": 600,
                  "zIndex": 1,
                }
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "height": "auto",
                    "opacity": 1,
                    "width": "auto",
                  }
                }
                testID="image-touchable-0"
              >
                <Image
                  source={
                    {
                      "uri": "mock-image-url",
                    }
                  }
                  style={
                    {
                      "borderRadius": 10,
                      "height": 300,
                      "resizeMode": "cover",
                      "width": "100%",
                    }
                  }
                />
              </View>
            </View>
          </View>
        </RCTScrollView>
        <Text
          numberOfLines={4}
          style={
            {
              "color": "black",
              "fontSize": 30,
              "fontWeight": "900",
              "margin": 20,
              "paddingRight": 10,
              "paddingTop": 50,
            }
          }
        >
          Test Post
        </Text>
        <Text
          style={
            {
              "color": "rgb(45,45,55)",
              "fontSize": 22,
              "fontWeight": "700",
              "margin": 20,
              "marginTop": 50,
            }
          }
        >
          This is a test post content
        </Text>
        <View
          style={
            {
              "alignItems": "flex-start",
              "backgroundColor": "rgb(230,230,230)",
              "borderTopLeftRadius": 20,
              "borderTopRightRadius": 20,
              "display": "flex",
              "flexDirection": "column",
              "justifyContent": "flex-start",
              "marginTop": 50,
              "padding": 20,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              {
                "color": "rgb(45,45,55)",
                "fontSize": 20,
                "fontWeight": "900",
              }
            }
          >
            Odpowiedzi
          </Text>
          <View>
            <View
              accessibilityRole="progressbar"
              accessibilityState={
                {
                  "busy": true,
                }
              }
              accessible={true}
              style={
                [
                  {
                    "alignItems": "center",
                    "justifyContent": "center",
                  },
                  {
                    "margin": 15,
                  },
                ]
              }
              testID="activity-indicator"
            >
              <View
                collapsable={false}
                style={
                  {
                    "height": 24,
                    "opacity": 1,
                    "width": 24,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "alignItems": "center",
                      "bottom": 0,
                      "justifyContent": "center",
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "height": 24,
                        "transform": [
                          {
                            "rotate": "45deg",
                          },
                        ],
                        "width": 24,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        {
                          "height": 12,
                          "overflow": "hidden",
                          "width": 24,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          {
                            "height": 24,
                            "transform": [
                              {
                                "translateY": 0,
                              },
                              {
                                "rotate": "-165deg",
                              },
                            ],
                            "width": 24,
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          style={
                            {
                              "height": 12,
                              "overflow": "hidden",
                              "width": 24,
                            }
                          }
                        >
                          <View
                            collapsable={false}
                            style={
                              {
                                "borderColor": "gray",
                                "borderRadius": 12,
                                "borderWidth": 2.4,
                                "height": 24,
                                "width": 24,
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  collapsable={false}
                  style={
                    {
                      "alignItems": "center",
                      "bottom": 0,
                      "justifyContent": "center",
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "height": 24,
                        "transform": [
                          {
                            "rotate": "45deg",
                          },
                        ],
                        "width": 24,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        {
                          "height": 12,
                          "overflow": "hidden",
                          "top": 12,
                          "width": 24,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          {
                            "height": 24,
                            "transform": [
                              {
                                "translateY": -12,
                              },
                              {
                                "rotate": "345deg",
                              },
                            ],
                            "width": 24,
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          style={
                            {
                              "height": 12,
                              "overflow": "hidden",
                              "width": 24,
                            }
                          }
                        >
                          <View
                            collapsable={false}
                            style={
                              {
                                "borderColor": "gray",
                                "borderRadius": 12,
                                "borderWidth": 2.4,
                                "height": 24,
                                "width": 24,
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <Text
              style={
                {
                  "color": "gray",
                  "fontSize": 18,
                  "fontWeight": "900",
                  "margin": 10,
                }
              }
            >
              Nic tu nie ma. Bądź pierwszy, który skomentuje!
            </Text>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "borderRadius": 20,
                  "height": 55,
                  "margin": 20,
                  "opacity": 1,
                  "width": 300,
                }
              }
            >
              <ViewManagerAdapter_ExpoLinearGradient
                colors={
                  [
                    4281150775,
                    4281150775,
                    4284769390,
                  ]
                }
                style={
                  {
                    "alignItems": "center",
                    "borderRadius": 20,
                    "display": "flex",
                    "height": "100%",
                    "justifyContent": "center",
                    "width": "100%",
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "white",
                      "fontSize": 22,
                      "fontWeight": "900",
                    }
                  }
                >
                  Odpowiedz
                </Text>
              </ViewManagerAdapter_ExpoLinearGradient>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
